### PR TITLE
[lldb][AIX] 1. Avoid namespace collision on other platforms

### PR DESCRIPTION
**Pull request by DhruvSrivastavaX**
This PR is in reference to porting LLDB on AIX.

Link to discussions on llvm discourse and github:
1.  https://discourse.llvm.org/t/port-lldb-to-ibm-aix/80640
2.  #101657 

The complete changes for porting are present in this draft PR:
https://github.com/llvm/llvm-project/pull/102601

The changes on this PR are intended to avoid namespace collision for certain typedefs between lldb and other platforms:
1. tid_t --> lldb::tid_t
2. offset_t --> lldb::offset_t

### CODE REVIEW

**Comment by DavidSpickett** [comment]
These seem fine but can you provide some details why this might have happened in your build and not in others?

Like:
* Are these tid_t and offset_t defined differently on AIX?
* Are you building with the system compiler or clang?

Include issues are not always easily explained, so if you can't, that's ok too. I would like to record the reasoning if we can though.

Probably best to be explicit about very common type names like this anyway.

**Comment by DavidSpickett** [comment]
Normally I would advise you to ignore the formatter for changes like this, but this is small enough and the intent will be clear even with the formatting changes. So please follow the guidance in the formatting comment to fix it.

**Comment by DavidSpickett** [comment]
And I edited the commit title slightly. The usual format is `[sub-project][area] <title>`, so if you could keep to that form. Not a hard rule but it is useful for skimming logs later.

**Comment by DavidSpickett** [comment]
And for the formatting, I usually use the clang-format in my llvm build dir. You don't have to go hunt down another specific version.

**Comment by DhruvSrivastavaX** [comment]
Ok Great, @DavidSpickett 
I was hoping for some feedback to get more clarity about llvm PRs.
1. Yes, the definition of tid_t and offset_t can differ in certain scenarios in AIX when compared to lldb, so trying to avoid such conflicts. 
2. We have used ibm-clang which is an AIX compatible version of clang, Here is the link:
https://www.ibm.com/docs/en/openxl-c-and-cpp-aix/17.1.1?topic=cc-highly-configurable-compiler
3. clang-format: Ok, thanks for that. I will use clang-format to avoid such formatting errors in the future.
4. [sub-project][area] <title> : Yes, thanks I had some doubt about that. Thanks for clarifying. Also, I am numbering these porting commits to keep additional tracking.

**Comment by DavidSpickett** [comment]
Assuming the formatting is fine I'll merge this tomorrow.

Before I do, please turn off private email addresses in your GitHub settings so that we record a proper commit email and you'll get buildbot notifications to the right place.

**Comment by DhruvSrivastavaX** [comment]
Great!!

Yes, I have turned off private email addresses now. 

Just a general question. Is it a one time thing or it is advisable to keep it off in future as well?

**Comment by DhruvSrivastavaX** [comment]
Hey David, 

The code format test passed but, 
I see some new failures (mostly in lldb-api and some in lldb-shell), which were not there in the previous run. 

Could it be because my branch is not up to date?

How can I avoid such failures for future PRs?

**Comment by DavidSpickett** [comment]
> The code format test passed but, I see some new failures (mostly in lldb-api and some in lldb-shell), which were not there in the previous run.

Not sure if you mean in the GitHub CI or locally, but either way I doubt they are due to your change. I'll see what the current CI run does.

> Could it be because my branch is not up to date?

Well, it could in fact be because your branch *is* up to date :) LLVM is a "live at head" sort of model where the main branch can be all kinds of broken depending on when you pull it. We fix these issues pretty fast but you can still be unlucky.

The advice is not to rebase the PR if you can avoid it: https://llvm.org/docs/GitHub.html#rebasing-pull-requests-and-force-pushes

But for example if you post a PR and the very first CI run has failures, it would be fine to rebase it before any significant review has taken place. Just to make it clear the issues are not your fault. Or if the comments were all addressed, etc, you'll get a feel for it over time.

> How can I avoid such failures for future PRs?

Don't rebase unless you have to. If when you land the PR things fail because some later change interacts with yours, that's fine that's what the post commit Buildbots are intended to catch.

The other side effect of llvm being "live at head" is that reverts are very common, so if they happen don't be discouraged, it happens to everyone. I wrote this up in more verbose terms in https://llvm.org/docs/MyFirstTypoFix.html#issues-after-landing-your-pr.

**Comment by DavidSpickett** [comment]
(and I will merge this once I see that CI is happy, or unhappy in a way not related to your changes)

**Review by DavidSpickett — APPROVED** [review]

**Comment by DhruvSrivastavaX** [comment]
Ok thats a relief to hear. 

> Not sure if you mean in the GitHub CI or locally

Yes I was referring to the Github CI right now. 

Great, The first PR is happily merged.
Thanks alot for all of the help and inputs!!

We can now get started with the rest of the PRs since I got some clarity now.

**Pull request merged**
